tests/Kiota.Builder.Tests/Writers/Go/CodeMethodWriterTests.cs

```diff
@@ -291,16 +291,20 @@ private void AddSerializationBackingStoreMethods()
     }
     private CodeClass AddUnionTypeWrapper()
     {
-        var complexType1 = root.AddClass(new CodeClass
+        var complexType1 = root.AddInterface(new CodeInterface
         {
             Name = "ComplexType1",
-            Kind = CodeClassKind.Model,
+            Kind = CodeInterfaceKind.Model,
+            OriginalClass = new CodeClass { Name = "ComplexType1", Kind = CodeClassKind.Model }
```

Comment:
It's possible I might be handling the mock here incorrectly.

Reply:
LGTM!